lib/modules/dosomething/dosomething_api/resources/campaign_resource.inc

```diff
@@ -274,8 +274,8 @@ function _campaign_resource_signup($nid, $values) {
     unset($values['northstar_id']);
   }
 
-  // If we pass either a Northstar ID or Drupal UID to `uid`, convert it to a real UID.
-  $values['uid'] = dosomething_user_convert_to_legacy_id($values['uid']);
+  // If a Drupal UID is passed to `uid`, convert it to the user's Northstar ID.
+  $values['uid'] = dosomething_user_convert_to_northstar_id($values['uid']);
```

Comment:
I'm not sure this will work as is, since [`dosomething_signup_create`](https://github.com/DoSomething/phoenix/blob/6e80ae77499c1fd8b30508c1bad7ab6e8dd6d192/lib/modules/dosomething/dosomething_signup/dosomething_signup.module#L255) expects the value provided in the `$uid` parameter here to be a UID stored in the database and [passes it along to `user_load`](https://github.com/DoSomething/phoenix/blob/6e80ae77499c1fd8b30508c1bad7ab6e8dd6d192/lib/modules/dosomething/dosomething_signup/dosomething_signup.module#L268).